chain/evm/client.go

```diff
@@ -158,6 +158,7 @@ type executeSmartContractIn struct {
 
 	chainID       *big.Int
 	gasAdjustment float64
+	txType        uint8
```

Comment:
This could use a doc comment. Ideally also instead of `uint8` it would be a go enum.

Reply:
I confirmed go-ethereum library uses uint8 for txType. So I will keep uint8.